### PR TITLE
Use more ES6 syntaxes in the real world example

### PR DESCRIPTION
**Pull request by ChrisJamesC**
Convert the async example files to using fat arrows, spread operators and stateless components.

This is a follow up to what I started in #1942 and #1945

References #1800

### CODE REVIEW

**Comment by ellbee** [comment]
Thanks!

The example is still working and code changes reflect those already merged in  #1942 and #1945

File: examples/real-world/src/containers/UserPage.js [hunk 1]
```diff
@@ -29,22 +22,17 @@ class UserPage extends Component {
...
+  handleLoadMoreClick = () => this.props.loadStarred(this.props.login, true)
 
-  renderRepo([ repo, owner ]) {
-    return (
-      <Repo repo={repo}
-            owner={owner}
-            key={repo.fullName} />
-    )
-  }
+  renderRepo = ([ repo, owner ]) => <Repo
```
Comment (gaearon) [inline examples/real-world/src/containers/UserPage.js]: This shouldn’t be autobound (not an event handler).
This is also hard to read for somebody new to this syntax.
Please format it the way it was before.

File: examples/real-world/src/reducers/paginate.js [hunk 2]
```diff
@@ -16,34 +15,35 @@ export default function paginate({ types, mapActionToKey }) {
...
+        return { ...state,
           isFetching: false,
           ids: union(state.ids, action.response.result),
           nextPageUrl: action.response.nextPageUrl,
           pageCount: state.pageCount + 1
-        })
+        }
       case failureType:
-        return merge({}, state, {
+        return { ...state,
```
Comment (gaearon) [inline examples/real-world/src/reducers/paginate.js]: Please put `...state` on the next line.
```js
return {
  ...state,
  stuff
}
```

File: examples/real-world/src/containers/UserPage.js (hunk 1 above)
Comment (ChrisJamesC) [inline examples/real-world/src/containers/UserPage.js]: Sure. Will do.

**Comment by gaearon** [comment]
I tweaked this do don't worry!

**Comment by ChrisJamesC** [comment]
@gaearon I don't. I'm really happy to get some advice so I can make my next PRs better suited to the conventions.

**Pull request merged**